circuit-runtime/src/commonMain/kotlin/com/slack/circuit/runtime/Navigator.kt

```diff
@@ -32,7 +32,11 @@ public interface Navigator {
    * val loginScreens = navigator.resetRoot(HomeScreen)
    * ```
    */
-  public fun resetRoot(newRoot: Screen): List<Screen>
+  public fun resetRoot(
+    newRoot: Screen,
+    saveState: Boolean = false,
+    restoreState: Boolean = false,
```

Comment:
I wonder if, should we make these true by default, we can make the backstack impl decide this if it has a max saved state capacity set to `0`

Reply:
I thought about this, but I think using stateful resets is more of a UX decision. For example, you would need to probably use `resetRoot(Onboarding, restoreState = false)` for onboarding, because it should be started from fresh.

False feels like the safer default, and people can opt-in for the use cases where it makes sense (top-level navigation changes mostly).

Reply:
Maybe these should be lambdas though, to make it easier to decide what the value should be at runtime. To make a decision you probably need to know at the current root screen, which isn't possible right now without access to the `BackStack` instance

```
public fun resetRoot(
    newRoot: Screen,
    saveState: (currentRoot: Screen) -> Boolean = { false },
    restoreState: (currentRoot: Screen) -> Boolean = { false },
)
```

Reply:
Let's do both - make resetRoot take a lambda and expose an extension that offers simple boolean params?

Reply:
Done!